src/ray/core_worker/test/direct_task_transport_test.cc

```diff
@@ -935,6 +935,69 @@ TEST(DirectTaskTransportTest, TestWorkerLeaseTimeout) {
   ASSERT_FALSE(raylet_client->ReplyCancelWorkerLease());
 }
 
+TEST(DirectTaskTransportTest, TestKillExecutingTask) {
+  rpc::Address address;
+  auto raylet_client = std::make_shared<MockRayletClient>();
+  auto worker_client = std::make_shared<MockWorkerClient>();
+  auto store = std::make_shared<CoreWorkerMemoryStore>();
+  auto factory = [&](const rpc::Address &addr) { return worker_client; };
+  auto task_finisher = std::make_shared<MockTaskFinisher>();
+  CoreWorkerDirectTaskSubmitter submitter(address, raylet_client, factory, nullptr, store,
+                                          task_finisher, ClientID::Nil(), kLongTimeout);
+  std::unordered_map<std::string, double> empty_resources;
+  ray::FunctionDescriptor empty_descriptor =
+      ray::FunctionDescriptorBuilder::BuildPython("", "", "", "");
+  TaskSpecification task = BuildTaskSpec(empty_resources, empty_descriptor);
+
+  ASSERT_TRUE(submitter.SubmitTask(task).ok());
+  ASSERT_TRUE(raylet_client->GrantWorkerLease("localhost", 1234, ClientID::Nil()));
+
+  // Try force kill, exiting the worker
+  ASSERT_TRUE(submitter.KillTask(task, true).ok());
+  ASSERT_TRUE(worker_client->ReplyPushTask(Status::IOError("workerdying"), true));
```

Comment:
This looks like there should probably be some check that we actually sent a `KillTask` request to the `worker_client`.

Reply:
Will do!